pylint/reporters/_json_reporter.py

```diff
@@ -0,0 +1,208 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+"""JSON reporter."""
+
+from __future__ import annotations
+
+import json
+import warnings
+from typing import TYPE_CHECKING, Optional, TypedDict
+
+from pylint.interfaces import CONFIDENCE_MAP, UNDEFINED
+from pylint.message import Message
+from pylint.reporters.base_reporter import BaseReporter
+from pylint.typing import MessageLocationTuple
+
+if TYPE_CHECKING:
+    from pylint.lint.pylinter import PyLinter
+    from pylint.reporters.ureports.nodes import Section
+
+# Since message-id is an invalid name we need to use the alternative syntax
+OldJsonExport = TypedDict(
+    "OldJsonExport",
+    {
+        "type": str,
+        "module": str,
+        "obj": str,
+        "line": int,
+        "column": int,
+        "endLine": Optional[int],
+        "endColumn": Optional[int],
+        "path": str,
+        "symbol": str,
+        "message": str,
+        "message-id": str,
+    },
+)
+
+
+class JSONReporter(BaseReporter):
+    """Report messages and layouts in JSON.
+
+    Consider using JSONReporterTwo instead, as this reporter has been deprecated.
+    """
+
+    name = "json"
+    extension = "json"
+
+    def display_messages(self, layout: Section | None) -> None:
+        """Launch layouts display."""
+        warnings.warn(
+            "The 'json' reporter has been deprecated in favour of a new version. "
+            "Consider switching to 'json2' which provides the same info in an updated "
+            "and improved format.",
+            DeprecationWarning,
```

Comment:
I don't think we have consensus to remove this just yet.
```suggestion
            PendingDeprecationWarning,
```

Reply:
👍  But let's keep the discussion going in the issue.